src/ws/remotectl-certificate.c

```diff
@@ -149,12 +157,24 @@ ensure_certificate (const gchar *user,
                     const gchar *group,
                     const gchar *selinux)
 {
+  char *error_str = NULL;
   GError *error = NULL;
   GTlsCertificate *certificate = NULL;
   gchar *path = NULL;
   gint ret = 1;
 
-  path = cockpit_certificate_locate (TRUE, &error);
+  path = cockpit_certificate_locate (&error_str);
+  if (error_str)
+    {
+      g_set_error_literal (&error, G_IO_ERROR, G_IO_ERROR_NOT_FOUND, error_str);
+      free (error_str);
+    }
+  if (path == NULL && g_error_matches (error, G_IO_ERROR, G_IO_ERROR_NOT_FOUND))
```

Comment:
The g_error_matches is redundant, no?  When cockpit_certificate_locate returns NULL, &error_str is always non-NULL.

Reply:
That's right, this can be simplified now.